DOC: add two tips for psql/pg_ctl DB management (#6344)
aiidateam/aiida-core: docs/source/intro/installation.rst

   $ psql

.. tip::

   If you see an error message like ``psql: FATAL:  role "<role_name>" does not exist``, probably you don't have any role created yet.
   You can use ``-d template1`` to connect to the default database and create a new role.

> 

Author: I use postgresql from conda, can anyone who use system postgresql test that `psql -d template1` will always work, then this tip is not need but can directly put `psql -d template1` as the command.

Reviewer: I can confirm that `psql -d template1` allows to connect as the `postgres` user when installed as a system wide service. However, there _are_ some roles present. I don't want to delete them all to check whether this would still work if no roles exist. But I think we can keep this and assume it works.

Reviewer: I don't really understand how this would fix the problem though. The error you mention I get for example when I try to run `psql` as my own user:
```
sph@invader:$ psql -d template1
psql: error: FATAL:  role "sph" does not exist
```
but that is because `psql` will use the linux system user if not explicitly specified, and my user does not have a correpsonding role to access. So how would switching the database to connect to fix that? As you can see, it doesn't for my example.

   $ psql

.. tip::

   If you see an error message like ``psql: FATAL:  role "<role_name>" does not exist``, probably you don't have any role created yet.

>    You can run ``psql -l`` to check the exist roles and specify the correct one with the ``-U`` option.

Reviewer: How does this part happen? Does postgres in Conda get installed without any roles?

Author: It happened when no role is specified, as the example you provided. 
This tip is for generic advance installation section, in `intro/installation.rst` not for conda. So it will happened and adding `-U` can help.

Reviewer: I see. But is the message than not incorrect? It is not so much that no role has been created yet, but that no role was specified?

Reviewer: The same goes for the second part of the tip about the database. By default it will connect to the `postgres` database. This should also always exist. Why would a user ever encounter the problem that this default database doesn't exist. How the hell did you encounter that?

Author: Because I use postgresql and I didn't run `sudo su - postgres`.

Reviewer: > Because I use postgresql 

Meaning? That is indeed what this section is about.

> and I didn't run `sudo su - postgres`

That is exactly my point. It is _that_ that is the problem. The tip should say that if you encounter the mentioned error, make sure to become the `postgres` system user.

Author: Sorry, I mean use postgresql from conda.

Author: If user follow the https://aiida.readthedocs.io/projects/aiida-core/en/stable/intro/install_conda.html, then it will encounter quite some issue in https://aiida.readthedocs.io/projects/aiida-core/en/stable/intro/installation.html. This is my point.

>    If you see an error message like ``psql: FATAL:  role "<role_name>" does not exist``, probably you don't have any role created yet.
>    You can run ``psql -l`` to check the exist roles and specify the correct one with the ``-U`` option.

Reviewer: You are using the term "role" here, but further down the language is to create a "user". I wonder if this is going to be obvious to users that they are equivalent
```suggestion
   If you see an error message like ``psql: FATAL:  role "<role_name>" does not exist``, probably you have not created any roles yet.
   You can run ``psql -l`` to check the existing roles and specify the correct one with the ``-U`` option.
```

>    If the error message is ``psql: FATAL:  database "<database_name>" does not exist``, you can use default database ``template1`` with ``psql -d template1``.

Reviewer: ```suggestion
   If the error message is ``psql: FATAL:  database "<database_name>" does not exist``, you can use the default database ``template1`` with ``psql -d template1`` to connect.
```

>    If you see an error message like ``psql: FATAL:  role "<role_name>" does not exist``, probably you have not created any roles yet.
>    You can run ``psql -l`` to check the existing roles and specify the correct one with the ``-U`` option.
>    If the error message is ``psql: FATAL:  database "<database_name>" does not exist``, you can use the default database ``template1`` with ``psql -d template1`` to connect.

Reviewer: ```suggestion
   If you have installed PostgreSQL through Conda and you see an error like ``psql: FATAL:  role "<role_name>" does not exist`` or ``psql: FATAL:  database "<database_name>" does not exist``, the default role and database apparently do no exist.
   The command ``psql -l`` prints the list of existing databases and the associated roles.
   You can try connecting to one of those by using the ``-d`` and ``-U`` option to specify the database and role, respectively, for example, ``psql -d template0 -U some-role``.
```
